db/external_sst_file_ingestion_job.cc

```diff
@@ -886,18 +886,29 @@ Status ExternalSstFileIngestionJob::AssignLevelAndSeqnoForIngestedFile(
     SequenceNumber* assigned_seqno) {
   Status status;
   *assigned_seqno = 0;
+  bool pick_l0_without_search = false;
   if (force_global_seqno) {
     *assigned_seqno = last_seqno + 1;
-    if (compaction_style == kCompactionStyleUniversal || files_overlap_) {
-      if (ingestion_options_.fail_if_not_bottommost_level) {
-        status = Status::TryAgain(
-            "Files cannot be ingested to Lmax. Please make sure key range of "
-            "Lmax does not overlap with files to ingest.");
-        return status;
-      }
+    if (compaction_style == kCompactionStyleUniversal) {
       file_to_ingest->picked_level = 0;
-      return status;
+      pick_l0_without_search = true;
+    }
```

Comment:
This is probably outdated. It looks like it was trying to uphold the pre-#7421 invariant. (btw, #12257 just removed another place we were trying to uphold that deleted invariant.)

Reply:
Thank you for this context. I have removed this default to L0 scenario and updated the unit test.